compiler/AST/iterator.cpp

```diff
@@ -412,14 +456,16 @@ buildZip1(IteratorInfo* ii, Vec<BaseAST*>& asts, BlockStmt* singleLoop) {
   }
 
   if (WhileStmt* stmt = toWhileStmt(singleLoop)) {
+    // By the time we get here, the condExpr has been passed through _cond_test
+    // and the result stored in a temp, so condExprForTmpVariableGet just
+    // returns the temp result.  So we can simply say:  (.= this more tmp)
+    // This assumes the backend will implicitly cast tmp to the type of more
```

Comment:
The assumption is verified for the C backend.  There is a checkbox in the pull request comment to ensure that it is also true for LLVM.  Both C and C++ specs are fairly explicit about "false" having an internal representation of 0 and "true" of 1.  You have to read a bit into the section on integral promotion to come to the conclusion that promoting a bool to int yields those same values, but that's my interpretation.  Any other behavior would be surprising.  ~hilde/Specifications/isoiec9899-tc3.pdf [6.3.1(2)].